### DISCUSSION TITLE
Should the discuss button load instantly?

### DISCUSSION THREAD

**Discussion by NoobTracker** [ScratchAddons/ScratchAddons#195] [Help]
Well ... one of the newest updates adds a thing for making scripts run eariler ... so should you be able to still see how it's loading?

**Comment by WorldLanguages** [comment]
> one of the newest updates adds a thing for making scripts run eariler

No. It adds a thing to making script run later, not earlier.

**Comment by NoobTracker** [comment]
What in the world (of languages bruh)
Wasn't the point of that update that things can be faster on the page?

**Comment by WorldLanguages** [comment]
No. There's no way to run modules before `defer` scripts are triggered.

**Comment by NoobTracker** [comment]
Wasn't that related to #159 ?

**Comment by WorldLanguages** [comment]
It is, so what?

**Comment by NoobTracker** [comment]
Well, there I asked if you can make scripts load earlier. And when the related PR just makes them load later then that's kinda not related.

**Comment by WorldLanguages** [comment]
It's related because DOMContentLoaded was mentioned, but that event actually happens after `defer`.

**Comment by NoobTracker** [comment]
So now I really don't understand anything. I just added that line to my extension since that didn't break it and I think that it now loads a lot faster. #196

**Comment by WorldLanguages** [comment]
It will load faster because the default value for `runAtComplete` is `true`, if you set it to false, it will load at `defer`, which was the previous default, and a bit faster than the current default.

**Comment by apple502j** [comment]
World_Languages seem to be mistaken. My PR adds a way to choose whether extensions waits for `load` (which was previous behavior, and current default) or not. Previously, script to load addon userscripts are always loaded after `load`, now it loads at `DOMContentLoaded` (but unless runAtComplete is false, delays importing userscripts until `load`)

**Comment by apple502j** [comment]
![image](https://user-images.githubusercontent.com/33279053/92238462-4842f500-eef4-11ea-8970-b49db585fbfe.png)

**Comment by WorldLanguages** [comment]
Yeah I admit this is confusing. But that should explain it ^
